### ISSUE TITLE
Normal Ladder melee seems to not work at all.

### ISSUE THREAD

**Issue by WC3Forever** [pvpgn/pvpgn-server#405]
I can connect from any PC on the LAN so far and test every seems good.
Except I cant play or join ladder games with any combination i try.

Also only 1 of the PC's is able to host with port 6112 the rest of the PCs cant host. Why is that? the old blizzard bnet could allow anyone to host all using the 6112 port. Why wont it work that way on the LAN with this server?

**Comment by cen1** [comment]
Did you set up w3route address_translation.conf correctly?

**Comment by WC3Forever** [comment]
i havent touched the address_translation file yet. I want to get everything working on the server first within the lan before making it publicly accessible. I would think ladder/melee would come working default.. not sure why its not. Do you have a server with it working? i'm using the newest version of the server out and only for wc3.

**Comment by cen1** [comment]
You say you connect on LAN but then you also say you need to port forward 6112 which makes no sense. Port forward is only needed for clients connecting on your external IP, not behind the NAT. If your PvPGN and other PC are on the same subnet (lets say 192.168.1.X) it should work out of the box afaik.

**Comment by WC3Forever** [comment]
Yeah i know but thats what i'm saying it doesn't work out of the box like that. I didnt say port-forward i only said hosting with port 6112 as in thats the port in the wc3 client on all PC's. I was saying when PCs are connection to a server over the internet they can all use port 6112 to connect to the old bnet server so i would think all my PCs on the LAN could also use the same 6112 port in the wc3 clients. I shouldn't have to change those wc clients on different PCs to different ports  within the LAN right?

Also what i was saying is 1 PC out of 4 is able to host the rest cant for some reason. No i tested with my sister from a different house last night and she is able to now connect to my server and host. I am also able to connect internally but unable to host myself.

Also ladder doesnt work at all in any combination and that is the main issue i'm trying to solve here.

**Comment by WC3Forever** [comment]
Hi Cen,

check this thread, is this true?
https://forums.pvpgn.pro/viewtopic.php?pid=11247#p11247

**Comment by cen1** [comment]
I will try to build on local machine and try to join ladder from LAN sometime this week. I usually deploy on servers and it works there for sure.

**Comment by WC3Forever** [comment]
How do you deploy from servers? During this corona social distancing some
of my friends need to play from home anyway. How would i made it so we
could play ladder together over the internet?
I have the server setup so they can join my server and ports forwarded, i
also have a ghost bot on my server to host with so its nicer. I just dont
know how to make it so we can play the actual ladder the way it was on old
classic bnet? with the dagger icon. And so it keeps score in our profile
and we earn icons ect..
Could you help me solve this part of it? Would be very grateful!

Also do you know if its possible to make playing a certain custom game can
also reward icon wins?




<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by cen1** [comment]
If your friend is joining your local pvpgn through port forwarding you most definitely need to configure address translation. As for deploying on server, it is exactly the same, you just need to know some linux basics and you buy a cheap VPS for a few $$ (can get them for less than 5€ a month).

**Comment by WC3Forever** [comment]
right now i'm just hosting this all on my server on some VM's. I have
plenty of bandwidth and cpu power and ssd storage and ram, so all good
there.
I'm running this on a windows server because my linux is so rusty..
my friend can join my server without port forwarding but obviously has to
port forward so they can host. I cant host yet though so i'll need to mess
with the address_translation for that soon. So far doesn't seem to work yet
because the instructions dont seem super clear on wording.

If you can help me get auto ladder working I would appreciate it a lot! I
could probably set something up and let you use a VM instance on my server
if needed for a while or something.

Also do you know if its possible to have certain custom games count for
icons like ladder does? i think i remember seeing something in the config
for that but not sure how it works.



<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by cen1** [comment]
One thing to clarify, your friend can join your pvpgn so that means you have port forwarded 6112 on your router to point to your pvpgn correct? If you port forwarded 6112->pvpgn then you can't host on 6112. Try to change your host port in game settings to 6113 and add 6113->your pc host . 

Then try to host a CG game and check if your ports are actually open for hosting, you cna use this tool: https://www.yougetsignal.com/tools/open-ports/

As for CG no, pvpgn does not track CG game sin any way.

**Comment by WC3Forever** [comment]
Yeah i have tried that and setup the address_translation file but doesn't
seem to work. my pc is ip 192.168.1.140 using port 6119 in wc still cant
host with anyone joining from internal or external. but i can join other
external people's games that host.

0.0.0.0:6200      my WAN ip:6200      192.168.1.0/24           ANY

192.168.1.140:6119   my WAN ip:6119      192.168.1.0/24,192.168.0.0/24  ANY

Also whats this setting mean?
# List additional game types to be counted as ladder games
# Curently allowed types: topvbot, melee, ffa, oneonone
# Example: ladder_games = "topvbot,oneonone" or use "none"
ladder_games = "none"

makes it seem like you can have some custom game names count as ladder for
icons?

Also
# If additional game types are configured (see above) to be counted as
ladder
# games then this setting configures a game name prefix to make only games
# which match this game name prefix be counted as ladder. This allows to
# still have normal games of the game types configured with "ladder_games"
# directive. However if this setting is commented or "" then ALL games
# which match the game types configured with "ladder_games" are to be
# considered as ladder games. The prefix checking is CASE SENSITIVE!
# Example: ladder_prefix = "ldr_"
ladder_prefix = ""

whats this setting mean?


<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by cen1** [comment]
This is as far as I could get it to work:

laptop 1: running pvpgn and w3 on 192.168.1.X
laptop 2: joined on mobile hotspot to emulate external client

1. Port forward 6112, 6200 and 6118
2. In address_translation.conf:

Expose w3route server:
0.0.0.0:6200      <your public ip>:6200      192.168.1.0/24           ANY

Expose your LAN client to external client:
<your LAN ip>:6118   <your public ip>:6118      192.168.1.0/24,10.0.0.0/8  ANY
127.0.0.1:6118	    <your public ip>:6118	     192.168.1.0/24 ANY

3. Open your w3, go to game settings and set host port to 6118
4. Enter bnet with both clients
5. Host a custom game
6. Verify that 6200 and 6228 is open with yougetsignal
7. Join CG game form client 2. Should work. In bnetd.log you should see that it translated 127.0.0.1 o your public IP
8. Now port forward client 2 and verify again with a CG game that it cna host
9. Now that both are verified to be able to host a CG game try a quickgame. You should be able to match and overcome the loading screen. With any luck, you should be able to start the game. If yo get disconnected after loading screen I am not sure anymore what the problem would be, I have not been able to figure out some of the address translation quirks.

**Comment by neel-bp** [comment]
@WC3Forever same problem happens with ghostbots too, the thing is locally pvpgn runs on 6112 ghostbots/ladder games work on that port too, so even locally you cannot run two things on same port you must change port for ladder/hostbots etc, so for testing i usually set hostport for bots to something other than 6112 and it works, might help you too, and i guess address translation is exactly for this.
